drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,958 @@
+		return no_os_gpio_set_value(desc->trig_desc, NO_OS_GPIO_HIGH);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Set channel state for specific channel
+ * @param desc - MAX22200 device descriptor.
+ * @param ch - Channel number.
+ * @param ch_state - Channel's state to be changed into.
+ * @return 0 in case of succces, negative error code otherwise.
+*/
+int max22200_set_ch_state(struct max22200_desc *desc, uint32_t ch,
+			  enum max22200_ch_state ch_state)
+{
+	if (ch > MAX22200_CHANNELS - 1)
+		return -EINVAL;
+
+	return max22200_reg_update(desc, MAX22200_STATUS_REG,
+				   MAX22200_ONCH_MASK(ch),
+				   no_os_field_prep(MAX22200_ONCH_MASK(ch),
+						   ch_state), false);
+}
+
+/**
+ * @brief Set fault mask bits in the status register.
+ * @param desc - MAX22200 device descriptor.
+ * @param fault_mask - Desired fault mask bit to set.
+ * @param enabled - enabled/disabled state of the mask.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_fault_mask_set(struct max22200_desc *desc,
+			    enum max22200_fault_mask fault_mask, bool enabled)
+{
+	if (fault_mask < 17 || fault_mask > 23)
+		return -EINVAL;
+
+	if (fault_mask == MAX22200_M_HHF && desc->chan_drive == MAX22200_VOLTAGE_DRIVE)
+		return -EINVAL;
+
+	return max22200_reg_update(desc, MAX22200_STATUS_REG,
+				   MAX22200_STATUS_FAULT_MASK(fault_mask),
+				   no_os_field_prep(MAX22200_STATUS_FAULT_MASK(fault_mask),
+						   enabled), false);
+}
+
+/**
+ * @brief Set device frequency value.
+ * @param desc - MAX22200 device descriptor.
+ * @param device_freq - Device frequency value.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_set_dev_freq(struct max22200_desc *desc,
+			  enum max22200_device_freq device_freq)
+{
+	if (device_freq > 1)
+		return -EINVAL;
+
+	return max22200_reg_update(desc, MAX22200_STATUS_REG,
+				   MAX22200_STATUS_FREQ_MASK,
+				   no_os_field_prep(MAX22200_STATUS_FREQ_MASK,
+						   device_freq), false);
+}
+
+/**
+ * @brief Set channel HIT time and HIT current.
+ * @param desc - MAX22200 device descriptor.
+ * @param ch - Channel for which HIT time is to be set.(0 based)
+ * @param hit_current - HIT current value, it shall not be greater than 127
+ * 			since the value is stored only on 7 bits in the
+ * 			channel's configuration register.
+ * @param hit_time - HIT time value to be set.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_set_ch_hit(struct max22200_desc *desc, uint32_t ch,
+			uint8_t hit_current, uint8_t hit_time)
+{
+	int ret;
+
+	if (ch > MAX22200_CHANNELS - 1)
+		return -EINVAL;
+
+	if (hit_current > MAX22200_HIT_MAX_VAL)
+		return -EINVAL;
+
+	ret = max22200_reg_update(desc, MAX22200_CFG_CH(ch),
+				  MAX22200_HIT_MASK,
+				  no_os_field_prep(MAX22200_HIT_MASK,
+						  (uint32_t)hit_current), false);
+	if (ret)
+		return ret;
+
+	return max22200_reg_update(desc, MAX22200_CFG_CH(ch),
+				   MAX22200_HIT_T_MASK,
```

Comment:
indentation is pretty odd. this applies to multiple returns throughout the code.